src/include/buffer.h

```diff
@@ -378,6 +378,88 @@ namespace buffer CEPH_BUFFER_API {
       }
     };
 
+    class appender {
+    protected:
+      list *pls;
+      size_t size;
+      ptr bp;
+      char *pos, *end;
+
+    public:
+      appender(bufferlist *l, size_t s) : pls(l), size(s) {
+	prepare_buffer();
+      }
+      appender(appender&& o)
+	: pls(o.pls), size(o.size),
+	  bp(std::move(o.bp)),
+	  pos(o.pos), end(o.end) {
+      }
+      ~appender() {
+	flush();
+      }
+
+      // no copying
+      appender(const appender& other) = delete;
+      appender& operator=(const appender& other) = delete;
+
+      void prepare_buffer(size_t at_least = 0) {
+	bp = buffer::create(std::max(size, at_least));
+	pos = bp.c_str();
+	end = pos + bp.length();
```

Comment:
@liewegas why call `bp.length()` here if it's `std::max(size, at_least)`? Just reuse the latter (could be that optimizer will be smart enough, but no need to)